wsmaster/che-core-sql-schema/src/main/resources/che-schema/7.0.0-beta4.0/1__add_devfile.sql

```diff
@@ -0,0 +1,256 @@
+
+---------
+
+-- devfile endpoint
+CREATE TABLE devfile_endpoint (
+    id                      BIGINT       NOT NULL,
+    name                    VARCHAR(255) NOT NULL,
+    port                    INTEGER      NOT NULL,
+    devfile_endpoints_id    BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_endpoint ADD CONSTRAINT fk_devfile_endpoint_id FOREIGN KEY (devfile_endpoints_id) REFERENCES devfile_component (id);
+CREATE INDEX index_devfile_endpoint_id ON devfile_endpoint (devfile_endpoints_id);
+
+-- devfile endpoint attributes
+CREATE TABLE devfile_endpoint_attributes (
+    endpoint_id    BIGINT,
+    value          TEXT,
+    name           VARCHAR(255)
+);
+-- constraints & indexes
+ALTER TABLE devfile_endpoint_attributes ADD CONSTRAINT fk_devfile_endpoint_attributes_id FOREIGN KEY (endpoint_id) REFERENCES devfile_endpoint (id);
+CREATE INDEX index_devfile_endpoint_attributes ON devfile_endpoint_attributes (endpoint_id);
+
+---------
+
+-- devfile component env
+
+CREATE TABLE devfile_env (
+    id                  BIGINT       NOT NULL,
+    name                VARCHAR(255) NOT NULL,
+    value               TEXT,
+    devfile_env_id      BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_env ADD CONSTRAINT fk_devfile_env_id FOREIGN KEY (devfile_env_id) REFERENCES devfile_component (id);
+CREATE INDEX index_component_env_id ON devfile_env (devfile_env_id);
+
+---------
+
+-- devfile component volume
+
+CREATE TABLE devfile_volume (
+    id                     BIGINT       NOT NULL,
+    name                   VARCHAR(255) NOT NULL,
+    container_path         TEXT,
+    devfile_volumes_id     BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_volume ADD CONSTRAINT fk_devfile_volumes_id FOREIGN KEY (devfile_volumes_id) REFERENCES devfile_component (id);
+CREATE INDEX index_component_volume_id ON devfile_volume (devfile_volumes_id);
+
+---------
+
+-- devfile component entrypoint
+
+CREATE TABLE devfile_entrypoint (
+    id                         BIGINT       NOT NULL,
+    parent_name                VARCHAR(255) NOT NULL,
+    container_name             VARCHAR(255) NOT NULL,
+    devfile_entrypoints_id     BIGINT,
+
+    PRIMARY KEY (id)
+);
+-- constraints & indexes
+ALTER TABLE devfile_entrypoint ADD CONSTRAINT fk_devfile_entrypoints_id FOREIGN KEY (devfile_entrypoints_id) REFERENCES devfile_component (id);
+CREATE INDEX index_component_entrypoint_id ON devfile_entrypoint (devfile_entrypoints_id);
+
+
+CREATE TABLE entrypoint_arg (
+    entrypoint_id    BIGINT,
+    args             VARCHAR(255)
+);
+
+-- constraints & indexes
+ALTER TABLE entrypoint_arg ADD CONSTRAINT fk_entrypoint_arg_id FOREIGN KEY (entrypoint_id) REFERENCES devfile_entrypoint (id);
+CREATE INDEX index_entrypoint_arg_entrypoint_id ON entrypoint_arg (entrypoint_id);
+
+
+CREATE TABLE entrypoint_commands (
+    entrypoint_id    BIGINT,
+    commands         VARCHAR(255)
```

Comment:
PRIMARY KEY (entrypoint_id)
entrypoint_id-> devfile_entrypoint_id

Reply:
PRIMARY KEY - not possible since it is not unique
field rename OK